.github/workflows/function_app_deploy.yaml

```diff
@@ -0,0 +1,147 @@
+on:
+  workflow_call:
+    inputs:
+      workspace_name:
+        type: string
+        required: true
+        description: The name of the workspace to create the artifact for.
+      environment:
+        description: Environment where the artifact will be deployed.
+        type: string
+        required: true
+      resource_group_name:
+        description: Function App resource group name.
+        type: string
+        required: true
+      function_app_name:
+        description: Function App name.
+        type: string
+        required: true
+      health_check_path:
+        description: The health probe path exposed by the Function App.
+        type: string
+        required: true
+      use_private_agent:
+        description: Use a private agent to deploy the built artifact.
+        type: boolean
+        required: false
+        default: true
+
+concurrency:
+  group: ${{ github.workflow }}-cd
+  cancel-in-progress: true
+
+env:
+  BUNDLE_NAME: bundle
+
+jobs:
+
+  build:
+    name: Build Artifact
+    runs-on: ubuntu-20.04
+    env:
+      WORKSPACE: ${{ inputs.workspace_name }}
+
+    steps:
+
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        name: Checkout
+
+      - name: Prune
+        run: npx turbo@1.9.3 prune --scope ${{ env.WORKSPACE }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          node-version-file: ".node-version"
+          cache: "yarn"
+          cache-dependency-path: "./out/yarn.lock"
+
+      - name: Install dependencies
+        run: yarn install --immutable
+        working-directory: ./out
+
+      - name: Restore turbo cache
+        uses: actions/cache@ab5e6d0c87105b4c9c2047343972218f562e4319 # v4.0.1
+        with:
+          path: ./out/node_modules/.cache/turbo
+          key: ${{ runner.os }}-turbo-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-turbo
+
+      - name: Build
+        run: yarn build
+        working-directory: ./out
+
+      - name: Build the Function App Artifact
+        id: make-function-app-artifact
+        run: |
+          npm pkg set --json "bundledDependencies"=true
+          npm pkg set --json "files"='["**/function.json", "dist", "host.json","extensions.csproj"]'
```

Comment:
do we need to copy `package.json` as well? @lucacavallaro

Reply:
`package.json` is always included by default :)